packages/mui-material/src/Slider/Slider.js

```diff
@@ -62,7 +63,7 @@ export const SliderRoot = styled('span', {
     },
     variants: [
       ...Object.entries(theme.palette)
-        .filter(([, palette]) => palette && palette.main)
+        .filter(createSimplePaletteValueFilter())
         .map(([color]) => ({
           props: { color },
           style: {
```

Comment:
I guess we could replace all of these variant generations with CSS variables? https://caniuse.com/?search=css%20variables is compatible with https://mui.com/material-ui/getting-started/supported-platforms/#browser. I guess this could mean for CSS static extraction a smaller CSS output. For emotion, I'm not sure it would bring a lot of benefits.